R/bundle.R

```diff
@@ -850,6 +850,13 @@ standardizeAppFiles <- function(appDir, appFiles = NULL, appFileManifest = NULL)
     # file list provided directly
     appFiles <- explodeFiles(appDir, appFiles)
   }
+
+  if (length(appFiles) == 0) {
+    cli::cli_abort(
+      "No content to deploy; {.arg appFiles} is empty"
```

Comment:
Could we adjust the message? This case could mean any of:

1. No files in the `appDir` directory to enumerate
2. No files in the manifest, or the files mentioned by the manifest do not exist
3. No files in `appFiles`, or the files included in `appFiles` do not exist.

Reply:
Makes sense. 

Since I'm in here, do you think I should make `appFiles` and `appFileManifest` mutually exclusive? It seems like supplying both is likely to be a mistake.

Reply:
The implementation indicates that they are not both used at once, so yes. I expect that this will eventually need revisiting to consider all of the file manifest, an incoming manifest.json, the manually specified list of files, and their interactions with ignore files.

Reply:
Not in this PR, but I think `standardizeAppFiles()` should also get the dependency discovery code that currently lives in `deployDoc()` (https://github.com/rstudio/rsconnect/blob/config-dir/R/deployDoc.R#L48-L54). That will eliminate the currently confusing `deployApp()` -> `deployDoc()` -> `deployApp()` sequence.